addons/web/static/src/views/list/list_renderer.xml

```diff
@@ -171,9 +171,9 @@
                 t-attf-class="o_group_name fs-6 fw-bold {{!group.isFolded ? 'text-900' : 'text-700'}}"
                 t-att-colspan="getGroupNameCellColSpan(group)">
                 <div class="d-flex align-items-center">
-                    <span t-attf-class="o_group_caret fa fa-fw {{group.isFolded ? 'fa-caret-right' : 'fa-caret-down' }} me-1"
+                    <span t-attf-class="o_group_caret fa {{group.isFolded ? 'fa-caret-right' : 'fa-caret-down' }}"
                         t-attf-style="--o-list-group-level: {{getGroupLevel(group)}}"/>
-                    <t t-esc="getGroupDisplayName(group)"/> (<t t-esc="group.count"/>)
+                        <span class="text-truncate ms-2 me-1" t-esc="getGroupDisplayName(group)"/> (<t t-esc="group.count"/>)
```

Comment:
The tabulation added is not needed, IMO.

Also, I am not sure we could do that in stable, let's see what Framework JS thinks.